addons/mail/models/mail_thread.py

```diff
@@ -4017,7 +4017,8 @@ def _message_update_content(self, message, body, attachment_ids=None,
         if strict:
             self._check_can_update_message_content(message.sudo())
 
-        msg_values = {'body': body} if body is not None else {}
+        msg_body = tools.html2plaintext(message.body)
+        msg_values = {'body': body, 'edit_date': fields.Datetime.now()} if msg_body != body else {}
```

Comment:
Continuing previous comment: when e modify attachments, you don't update "edit_date", because it is done in a low level method ... this is a bit ping-pong code, which is hard to follow. Feel like the main method being this one, it should always be called through this one and updated in this one.